src/lib/crypto/krb/string_to_key.c

```diff
@@ -51,8 +51,11 @@ krb5_c_string_to_key_with_params(krb5_context context, krb5_enctype enctype,
         return KRB5_BAD_ENCTYPE;
     keylength = ktp->enc->keylength;
 
+    if (salt == NULL)
```

Comment:
Maybe collapse this into the next check?  `if (!salt || salt->length == SALT_TYPE_AFS_LENGTH)` or so?